proposals/lip-0051.md

```diff
@@ -769,14 +770,222 @@ if the above send function does not fail:
     # Notice that the forwardedMessageFee was deducted by the send function
 ```
 
-### Tokens and Genesis Blocks
 
-The genesis block of a chain can have a non-empty token store. The distribution of tokens at genesis is left to sidechain developers and must follow the conditions below:
+### Genesis Block Processing
+
+
+#### Genesis Assets Schema
+
+```java
+genesisTokenStoreSchema = {
+    "type": "object",
+    "required": [
+        "userSubstore",
+        "supplySubstore",
+        "escrowSubstore",
+        "availableLocalIDSubstore",
+        "terminatedEscrowSubstore"
+    ],
+    "properties": {
```

Comment:
the field numbers of these properties have to be fixed

Reply:
good catch, fixed (hope I did not miss any 😅 )